### ISSUE TITLE
Github tables without a header row

### ISSUE THREAD

**Issue by vanboom** [xijo/reverse_markdown#87]
I have some tables that are coded in HTML without <th> like this...

```html
<table>
<tr>
<td><b> header 1 </b></td> <td> <b> header 2 </b> </td>
<td> item 3 </td> <td> item 4 </td>
</tr>
</table>
```

```markdown
| header 1 | header 2 |
| item 3 | item 4 |
```

Suggestion:  `github_flavored` should treat the first row as the header row to match the Github markdown spec for tables.  The markdown shown above is not being converted back to HTML properly by Redcarpet or the github_markup gem (Ruby).   Thanks!

**Comment by xijo** [comment]
Hi @vanboom,

The github markdown specs looks no different to the regular one concerning tables: https://guides.github.com/features/mastering-markdown/

Maybe I don't get the point yet, but I'd rather not treat `td` with `b` tags any special unless it's a real thing in any spec.

**Comment by vanboom** [comment]
Yes, I agree that the td + b approach would not be the best option.  The issue is that I have a table that does not have a <th> in the header so the markdown is appearling like this:
```markdown
| header 1 | header 2 |
| item 3 | item 4 |
```
Without the `| ---- | ---- |` line to denote the header row, the Github markdown translator is not treating the table as a table when attempting to render the markdown back to HTML.   I am using the [github-markdown](https://github.com/github/markup) gem to convert the markdown to HTML.

**Comment by vanboom** [comment]
I was able to get a possible solution working on my fork which detects and treats the first row of the table as the header row regardless of the presence of <th>.  I will try to get a PR together for your consideration.
